vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.html

```diff
@@ -24,309 +27,339 @@
+//link icons for module links
+var moduleLinkIconReader = new FileReader();
+var moduleLinkIcon;
+moduleLinkIconReader.onload = function(e) {
+	moduleLinkIcon = e.target.result;
+    showModuleLink(createModuleLinkInfo());
 }
 
-function makeExternalLinksEditable(linkName, linkId, externalLinkURL, posXEdit, posYEdit, displayLinkId, linkType) {
-    selectedExternalLinkId=linkId;
-    storeX=posXEdit;
-	storeY=posYEdit;
-    $("#externalLinkInfoLabel").text(linkName);
-    $("#externalLinkId").val(linkId);
-	$("#externalLinkDisplayId").val(displayLinkId);
-	$("#externalLinkInfoLabelEdit").text(linkName);
-	$("#externalLinkLabelEdit").val(linkName);
-	$("#externalLinkIdValueEdit").val(linkId);
-	$("#externalLinkURLEdit").val(externalLinkURL);
-	$('#extTypeEdit option[value="'+linkType+'"]').attr("selected", "selected");
-    resetHighlighting();
-          
-    $('[data-link-id="' + linkId + '"]').css("color", "#c1bb88");
-    $('div[data-link-id="' + linkId + '"]').removeClass("alert-primary");
-    $('div[data-link-id="' + linkId + '"]').addClass("alert-warning");
-    $('img[data-link-id="' + linkId + '"]').css("border", "solid 1px #c1bb88");
-    
-    $("#bgImage").on("click", function(e){
-		e.preventDefault();
-		var posX = $(this).position().left;
-		var posY = $(this).position().top;    
-		storeX = e.pageX - $(this).offset().left;
-		storeY = e.pageY - $(this).offset().top;
-		showExternalLinkEdit(editExternalLinkInfo());
-	});
-    hideLinkInfoTabs();
-    $("#externalLinkInfo").show();
-    $("#editExternalLinkInfo").show();
+//link icons for external links
+var externalLinkIconReader = new FileReader();
+var externalLinkIcon;
+externalLinkIconReader.onload = function(e) {
+	externalLinkIcon = e.target.result;
+    showExternalLinks(createExternalLinkInfo());
 }
 
-function resetHighlighting() {
-    // reset icon links
-    $('[data-link-id]').css("color", "red");
-     
-    // reset alert links
-    $('div[data-link-id]').removeClass("alert-warning");
-    $('div[data-link-id]').addClass("alert-primary");
-    
-    // reset image links
-    $('img[data-link-id]').css("border-width", "0"); 
+function drawLinks(){
+    $("#main-space span").remove();
+	$("#main-space i").remove();
+	$("#space span").remove();
+	$("#space i").remove();
+    $.ajax({
+		type: "GET",
+		 url: "[(@{'/staff/space/'+${space.id}+'/links?'+${_csrf.parameterName}+'='+${_csrf.token}})]",
+		cache       : false,
+        contentType : false,
+        processData : false,
+        success: function(data) {
+        	var spaceLinks = data['spaceLinks'];
+        	var externalLinks = data['externalLinks'];
+        	var moduleLinks = data['moduleLinks'];
+        	var posXModal = $("#bgImage").position().left;
+        	var posYModal = $("#bgImage").position().top;
+        	var posX = $("#main-bgImage").position().left;
+        	var posY = $("#main-bgImage").position().top;
+        	var img = document.getElementById("bgImage");

```

Comment:
please don't mix jquery and regular js expressions (besides when it's absolutely necessary)